src/OVAL/probes/unix/sysctl_probe.c

```diff
@@ -268,6 +280,76 @@ int sysctl_probe_main(probe_ctx *ctx, void *probe_arg)
 
         return (0);
 }
+
+#elif defined(OS_FREEBSD)
+int sysctl_probe_main(probe_ctx *ctx, void *probe_arg)
+{
+        FILE *fp;
+        char output[LINE_MAX];
+        const char* SEP = "=";
+        char* mib;
+        char* sysval;
+        SEXP_t *se_mib;
+        SEXP_t *name_entity, *probe_in;
+
+        probe_in    = probe_ctx_getobject(ctx);
+        name_entity = probe_obj_getent(probe_in, "name", 1);
+
+        if (name_entity == NULL) {
+                dE("Missing \"name\" entity in the input object");
+                return (PROBE_ENOENT);
+        }
+
+        fp = popen(SYSCTL_CMD, "r");
```

Comment:
Is there a way to fetch this information without spawning another process? While it is not being a *hard* restriction, we would prefer not to spawn anything during the probe execution. So far we were successful in avoiding it.

Reply:
I'm not entirely happy with having to call `/sbin/sysctl` either, but I am not aware of a better solution at the moment. I looked at the source code of FreeBSD's `sysctl` to see how it is obtaining this information, and saw that it uses undocumented, and potentially unstable, kernel interfaces [1] to accomplish this.

[1] https://cgit-beta.freebsd.org/src/tree/sbin/sysctl/sysctl.c#n862